jlh/JonFirstApp/app/src/main/java/com/e_conomic/jonfirstapp/MainFragment.java

```diff
@@ -184,6 +174,44 @@ public void onClick(View view) {
         });
     }
 
+    /** Sends the user entered message if permission was granted. Requests the permission if the app
+     * does not already have it. Also shows a dialog if now message was entered. */
+    private void sendMessage() {
+
+        message = editMessage.getText().toString();
+
+        if (message.length() <= 0) {
+            EnterMessageDialogFragment dialog = new EnterMessageDialogFragment();
+            dialog.show(getFragmentManager(), ENTER_MESSAGE_FRAGMENT_TAG);
+            return;
+        }
+
+        // If the user does not wish to send an SMS.
+        if (contactPhoneNumber == null) {
```

Comment:
Not sure I understand the comment and logic here. If no `contactPhoneNumber` is entered, does it mean what the comment says? Or rather, is it that the `contactPhoneNumber` is simply missing? As in, the user forgot to enter it? 

Either way, I would create a method wrapping the logic and meaning, which would make the code self-explanatory, and hide away details. E.g. a method called `private bool phoneNumberHasBeenEntered()`